File: tests/nodejs/cases/case_139_serialized_copy_only_04.js
/**
 * Task case 139: serialized_copy_only_04 false-positive/false-negative stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case139SerializedCopyOnly04(input) {
  const taskName = 'serialized_copy_only_04';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'stress',
    stage: 'evaluation',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const serialized = JSON.stringify(payload);
  // SAFE: only primitive serialized data leaves scope.
  return serialized;
}

module.exports = {
  case139SerializedCopyOnly04,
};

File: tests/rust/cases/case_139_serialized_copy_only_04.rs
use std::collections::HashMap;
use std::sync::{Mutex, OnceLock};

static RETAINED_CACHE: OnceLock<Mutex<Vec<HashMap<String, String>>>> = OnceLock::new();
static RETAINED_AUDIT: OnceLock<Mutex<Vec<HashMap<String, String>>>> = OnceLock::new();
static RETAINED_HANDLERS: OnceLock<Mutex<Vec<String>>> = OnceLock::new();

pub fn case_139_serialized_copy_only_04(input: String) -> String {
    let task_name = "serialized_copy_only_04".to_string();
    let raw = if input.is_empty() { "sample".to_string() } else { input };
    let mut payload: HashMap<String, String> = HashMap::new();
    payload.insert("task".to_string(), task_name.clone());
    payload.insert("entity".to_string(), "stress".to_string());
    payload.insert("stage".to_string(), "evaluation".to_string());
    payload.insert("input".to_string(), raw.clone());
    payload.insert("checksum".to_string(), format!("{}:{}", task_name, raw.len()));
    let serialized = format!(
        "{}:{}",
        payload.get("task").cloned().unwrap_or_default(),
        payload.get("input").cloned().unwrap_or_default()
    );
    // SAFE: only serialized string escapes.
    serialized
}
